### DISCUSSION TITLE
How to activate render markdown automatically

### DISCUSSION THREAD

**Discussion by andtankian** [olimorris/codecompanion.nvim#456] [Q&A] [answered]
@olimorris 

It seems in your examples that you use [render-markdown](https://github.com/MeanderingProgrammer/render-markdown.nvim). 
Can you share a couple of config with us? I mean, in my case I need to enter in the chat buffer and `set ft=markdown` so I get it working (despite the fact the `cmp` for` /slash` commands stop work)
Is there any extra config we need to do?
(I also already disabled header rendering as suggested in the documentation) but no success.

**Comment by olimorris** [comment, marked as answer]
Hey @andtankian this is my setup with Lazy:

```lua
{
    "MeanderingProgrammer/render-markdown.nvim", -- Make Markdown buffers look beautiful
    ft = { "markdown", "codecompanion" },
    opts = {
      render_modes = true, -- Render in ALL modes
      sign = {
        enabled = false, -- Turn off in the status column
      },
    },
  },

```

**Reply by rbhanot4739** [reply]
@olimorris sorry to comment on this old thread, but how did you highlight those variable `#buffer` and tool  (`@editor`) sections as shown in the videos ?

**Reply by olimorris** [reply]
The plugin creates [highlight groups](https://codecompanion.olimorris.dev/configuration/others.html#highlight-groups). The theme I use tweaks them a little.

**Reply by rbhanot4739** [reply]
Yeah I was checking that in the docs however even with very minimal setup these don't seem to get highlighted for me.

```lua
  {
    "olimorris/codecompanion.nvim",
    enabled = true,
    opts = {},
    dependencies = {
      "nvim-lua/plenary.nvim",
      "nvim-treesitter/nvim-treesitter",
      "ravitemer/codecompanion-history.nvim",
      {
        "MeanderingProgrammer/render-markdown.nvim", -- Enhanced markdown rendering
        dependencies = { "nvim-treesitter/nvim-treesitter", "nvim-tree/nvim-web-devicons" },
        ft = { "markdown", "codecompanion" },
  },
}
```
<img width="858" alt="image" src="https://github.com/user-attachments/assets/d200416e-7b74-4440-afd4-fb355cc0c872" />

**Reply by olimorris** [reply]
Yeah that looks correct. This isn't related to render-markdown at all btw

**Comment by Felix-Paetsch** [comment]
If someone else stumbles upon the same problem:
It seems like the latest version of render-markdown uses the `file_types` key now instead of `ft`.
